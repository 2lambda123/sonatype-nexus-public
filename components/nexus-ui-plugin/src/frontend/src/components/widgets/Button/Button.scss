/**
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */

@import '../../../styles/colors';

.nx-btn {
  background-color: $almost-white;
  border: solid 1px $almost-black;
  border-radius: 3px;
  color: $almost-black;
  font-family: inherit;
  font-size: 1em;
  height: 2em;
  min-width: 76px;
  opacity: 1;
  cursor: pointer;

  &:hover:not(:disabled) {
    filter: brightness(0.9);
  }

  &:disabled {
    cursor: default;
    opacity: 0.5;
  }

  &:focus, &:active {
    border-color: $light-blue;
  }

  &.nx-btn--primary {
    background-color: $blue;
    color: $white;
  }
}
