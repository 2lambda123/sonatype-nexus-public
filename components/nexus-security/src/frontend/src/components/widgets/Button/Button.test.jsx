/*
 * Sonatype Nexus (TM) Open Source Version
 * Copyright (c) 2008-present Sonatype, Inc.
 * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
 *
 * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
 * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
 *
 * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
 * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
 * Eclipse Foundation. All other trademarks are the property of their respective owners.
 */
import {shallow} from 'enzyme';
import React from 'react';

import Button from './Button';

describe('Button', () => {
  const getButton = (extraProps) => {
    let props = {
      name: 'b1',
      onChange: () => {
      },
      ...extraProps
    };
    return shallow(<Button {...props} />);
  };

  it('renders correctly', () => {
    expect(getButton()).toMatchSnapshot();
  });

  it('renders correctly as a primary button', () => {
    expect(getButton({isPrimary: true})).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    expect(getButton({disabled: true})).toMatchSnapshot();
  });

  it('allows for its style to be overriden', () => {
    expect(getButton({
      style: {
        border: 'none'
      }
    })).toMatchSnapshot();
  });
});
