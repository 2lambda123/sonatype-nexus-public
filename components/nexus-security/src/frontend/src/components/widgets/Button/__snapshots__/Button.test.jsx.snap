// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button allows for its style to be overriden 1`] = `
<button
  className="nxrm-button"
  name="b1"
  onChange={[Function]}
  style={
    Object {
      "border": "none",
    }
  }
/>
`;

exports[`Button renders correctly 1`] = `
<button
  className="nxrm-button"
  name="b1"
  onChange={[Function]}
/>
`;

exports[`Button renders correctly as a primary button 1`] = `
<button
  className="nxrm-button nxrm-button-primary"
  name="b1"
  onChange={[Function]}
/>
`;

exports[`Button renders correctly when disabled 1`] = `
<button
  className="nxrm-button"
  disabled={true}
  name="b1"
  onChange={[Function]}
/>
`;
